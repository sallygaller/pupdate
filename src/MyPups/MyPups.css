.MyPups {
    text-align: center;
    display: flex;
    width: 50%;
    margin: 0 auto;
    flex-direction: column;
    justify-content: center;
}

.MyPups h2 {
    margin-bottom: 10px;
}

.MyPups ul {
    padding: 0px;
}

.MyPups-img {
    max-width: 350px;
    margin: 0 auto;
}

.MyPups-profile {
    width: 25%;
    background-color: #C28832;
    color: white;
    margin: 10px;
    padding: 10px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}

.MyPups-delete-profile {
    width: 25%;
    background-color: #B31E00;
    color: white;
    margin: 10px;
    padding: 10px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}

.MyPups-add {
    width: 25%;
    background-color: #297537;
    color: white;
    margin: 5px;
    padding: 10px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}