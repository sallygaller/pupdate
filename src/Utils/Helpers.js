export default function playstyles(playstyle) {
  if (playstyle === "nervous") {
    return "I'm nervous or shy around other dogs.";
  }
  if (playstyle === "gentle") {
    return "I play gently.";
  }
  if (playstyle === "food-obsessed") {
    return "I'm food-obsessed!";
  }
  if (playstyle === "walks") {
    return "I like going on walks with my pup pals.";
  }
  if (playstyle === "parks") {
    return "I like going to dog parks with my pup pals.";
  }
  if (playstyle === "wrestling") {
    return "I like playfighting and wrestling.";
  }
  if (playstyle === "rambunctious") {
    return "I'm rambunctious and playful.";
  }
  if (playstyle === "ball-obsessed") {
    return "I love to play fetch!";
  }
}
