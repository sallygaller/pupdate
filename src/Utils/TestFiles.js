export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
