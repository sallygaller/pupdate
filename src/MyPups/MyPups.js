import React from "react";
import { Link } from "react-router-dom";
import { pupImage } from "../Utils/Helpers";
import "./MyPups.css";

export default function MyPups(props) {
  const pups = props.pups;
  return (
    <div className="MyPups">
      <h2>My Pups</h2>
      <Link to={`/addpup`}>
        <button className="MyPups-add" type="button">
          Add a Pup!
        </button>
      </Link>
      <ul>
        {pups.map((pup) =>
          pup.id === 1 ? (
            <li key={pup.id}>
              <h3>{pup.name}</h3>
              <img
                className="MyPups-img"
                alt={pup.name}
                src={pupImage(pup.id)}
              />
              <br></br>
              <Link to={`/pups/${pup.id}`}>
                <button className="MyPups-profile" type="button">
                  {pup.name}'s Play Profile
                </button>
              </Link>
            </li>
          ) : null
        )}
      </ul>
    </div>
  );
}
