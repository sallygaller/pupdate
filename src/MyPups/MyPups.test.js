import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import MyPups from "./MyPups";
import { pups } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <MyPups pups={pups} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MyPups pups={pups} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
