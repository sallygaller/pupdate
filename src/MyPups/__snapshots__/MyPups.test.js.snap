// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="MyPups"
>
  <h2>
    My Pups
  </h2>
  <a
    href="/addpup"
    onClick={[Function]}
  >
    <button
      className="MyPups-add"
      type="button"
    >
      Add a Pup!
    </button>
  </a>
  <ul />
</div>
`;
